### PR TITLE
(+semver: breaking) Updates to docker testing functions to support port forwarding

### PR DESCRIPTION
**Pull request by thzinc**
This allows the docker test functions to work on macOS and Windows where the networking works differently.

(Work done as a part of EN-5506)

### CODE REVIEW

**Review by dkaminski — APPROVED** [review]
LGTM

**Review by cuevaskoch — APPROVED** [review]
Test run on linux looks good, all LGTM :+1: 

```
❯ go test ./... 
?   	github.com/syncromatics/go-kit/database	[no test files]
?   	github.com/syncromatics/go-kit/grpc	[no test files]
?   	github.com/syncromatics/go-kit/internal/protos/gokit/ping/v1	[no test files]
?   	github.com/syncromatics/go-kit/log	[no test files]
?   	github.com/syncromatics/go-kit/redis	[no test files]
=== RUN   ExampleSetupEtcd
--- PASS: ExampleSetupEtcd (9.71s)
=== RUN   ExampleSetupKafka
--- PASS: ExampleSetupKafka (22.85s)
=== RUN   ExampleSetupMSSqlDatabase
--- PASS: ExampleSetupMSSqlDatabase (6.99s)
=== RUN   ExampleSetupPostgresDatabase
--- PASS: ExampleSetupPostgresDatabase (4.58s)
=== RUN   ExampleSetupRabbitMQ
--- PASS: ExampleSetupRabbitMQ (21.99s)
=== RUN   ExampleSetupRedis
--- PASS: ExampleSetupRedis (5.96s)
PASS
ok  	github.com/syncromatics/go-kit/testing/docker	72.091s

```

File: testing/docker/etcd_setup.go [hunk 1]
```diff
@@ -101,11 +126,30 @@ func removeEtcdContainer(client *client.Client, testName string) {
...
+func waitForEtcdToBeReady(client *client.Client, id string, grpcPort, metricsPort int) (string, error) {
+	healthURL := fmt.Sprintf("http://localhost:%d/health", metricsPort)
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+
+	for {
+		select {
+		case <-ctx.Done():
+			cancel()
+			return "", errors.New("failed to start container in a timely manner")
+		case <-time.After(1*time.Second):
```
Comment (annymsMthd) [inline testing/docker/etcd_setup.go:138]: Oh. This is nice. I didn't know about .After

**Comment by thzinc** [comment]
@cuevaskoch That's stellar! Thanks for confirming.

**Review by annymsMthd — APPROVED** [review]

File: testing/docker/etcd_setup.go (hunk 1 above)
Comment (thzinc) [reply testing/docker/etcd_setup.go:138]: I've been learning 😉

**Pull request merged**